lib/child_process.js

```diff
@@ -250,45 +245,43 @@ exports.execFile = function(file /*, args, options, callback*/) {
   }
 
   if (child.stdout) {
-    stdoutState = child.stdout._readableState;
+    if (encoding)
+      child.stdout.setEncoding(encoding);
 
     child.stdout.addListener('data', function(chunk) {
-      // If `child.stdout.setEncoding()` happened in userland, convert string to
-      // Buffer.
-      if (stdoutState.decoder) {
-        chunk = Buffer.from(chunk, stdoutState.decoder.encoding);
-      }
-
-      stdoutLen += chunk.byteLength;
+      stdoutLen += chunk instanceof Buffer ?
+        chunk.length :
+        Buffer.byteLength(chunk, encoding);
```

Comment:
Can you at least line these two lines up with `chunk`? Same for the other instance of this below.